amisr-user/resolvedvelocities
# Major refactoring of vvels

This is a major refactoring of the original vvels code.  The new version of the code is now `ResolveVectors.py`, which can be run with `test_resolve_vectors.py`.  Emphasis was on modularizing the code, implementing Apex magnetic coordinates, and removing bits of code that were unnecessary.  There are still a number of features that need to be added (plotting, run options, ect), but this contains the algorithm that actually calculates the vectors.

To test:
1. Fetch the refactor branch in your local repo.
```
git fetch
git checkout refactor
```
2. Enter the `test` directory.
```
cd resolvedvelocities/resolvedvelocities/tests
```
3. Modify the parameters in `example_config.ini` to match your file paths/how you want to run the code.  By and large, the default parameters in this file are taken directly from the original vvels code and should be representative of standard processing.  If `INTTIME` is not included, vvels will be calculated for each time step in the original fitted file (no post-integration).  If `USE_BEAMS` is not included, all beams will be used.
4. Run the code
```
python test_resolve_vectors.py example_config.ini
```
This creates the output file `test_vvels.h5`.  Obviously, this is not a good name for the output file, but it should include MOST of the relevant arrays (exact format of the output file still needs some work).  There's still a lot of work to, but let's make sure we're all happy with the general architecture of the code.  The aim is to make sure this is relatively easy to modify in the future (new magnetic field model, new binning, ect.), so please suggest recommendations!
